tests/test_telemetry.py

```diff
@@ -324,22 +322,20 @@ def test_app_heartbeat(self):
         # first heartbeat from the list
         heartbeats.pop(0)
 
-        for data in heartbeats:
-            curr_message_time = datetime.strptime(data["request"]["timestamp_start"], fmt)
-            if prev_message_time is None:
-                logger.debug(f"Heartbeat in {data['log_filename']}: {curr_message_time}")
-            else:
-                delta = (curr_message_time - prev_message_time).total_seconds()
-                logger.debug(f"Heartbeat in {data['log_filename']}: {curr_message_time} => {delta}s ellapsed")
+        timestamps = [datetime.strptime(data["request"]["timestamp_start"], fmt) for data in heartbeats]
 
-                assert (
-                    delta < UPPER_LIMIT
-                ), f"Heartbeat sent too slow ({delta}s). It should be sent every {expected_heartbeat_interval}s"
-                assert (
-                    delta > LOWER_LIMIT
-                ), f"Heartbeat sent too fast ({delta}s). It should be sent every {expected_heartbeat_interval}s"
+        # In theory, it's sorted. Let be safe
+        timestamps.sort()
 
-            prev_message_time = curr_message_time
+        delta = (timestamps[-1] - timestamps[0]).total_seconds()
+        average_delay = delta / (len(heartbeats) - 1)
```

Comment:
NIT: I think it would be clearer to do `len(timestamps)` instead of `len(heartbeats) - 1` (assuming that's legal in Python). It took me a bit of time to understand where the -1 came from.

Reply:
it'll be also `len(timestamps) -1`, it's `timestamps`, not `delays` :)

Reply:
Oooh you're right, my bad

Reply:
added a comment to make maths crystal clear